modules/csvparser/filterx-func-parse-csv.c

```diff
@@ -88,6 +88,27 @@ _parse_list_argument(FilterXFunctionParseCSV *self, FilterXExpr *list_expr, GLis
   return result;
 }
 
+static inline void
+_init_scanner(FilterXFunctionParseCSV *self, GList *string_delimiters, GList *cols, const gchar *input,
+              CSVScanner *scanner, CSVScannerOptions *local_opts)
+{
+  CSVScannerOptions *opts = &self->options;
+
+  if (string_delimiters || cols)
+    {
+      csv_scanner_options_copy(local_opts, &self->options);
+      opts = local_opts;
+    }
+
+  if (string_delimiters)
+    csv_scanner_options_set_string_delimiters(local_opts, string_delimiters);
+
+  if (cols)
+    csv_scanner_options_set_expected_columns(local_opts, g_list_length(cols));
```

Comment:
that g_list_length() is going to bite us, but I guess that's coming in a separate PR.

Reply:
Let's fix it here :)

Reply:
Fixed.